### DISCUSSION TITLE
Not connected to the database

### DISCUSSION THREAD

**Discussion by jeremyikwuje** [planetscale/discussion#234] [Q&A] [answered]
The connection failed after successfully entering my database details.

HOST=xxx
DATABASE=php_example
USERNAME=xxx
PASSWORD=pscale_pw_xx
MYSQL_ATTR_SSL_CA=

I didn't include the MYSQL_ATTR_SSL_CA, is that the reason why it didn't connect? My environment is Windows how do I get SSL_CA?

**Comment by fooforge** [comment, marked as answer]
@ijsucceed What error message did you get?

Aside from that, to connect to PlanetScale securely, your client needs to be able to validate our SSL/TLS certificate signed by one of the common certificate authorities. Otherwise, the connection will either fail or it will be susceptible to man-in-the-middle attacks.

For Windows, we recommend to download Mozilla's CA certificate bundle maintained by the curl project at https://curl.se/docs/caextract.html and to point your database driver to it. For more in-depth information on this topic please also see [our documentation](https://docs.planetscale.com/concepts/secure-connections).

**Reply by Sophyia7** [reply]
I have downloaded the SSL certificate and I don't know how to connect it to my `settings.py`. I added it to my root directory, though.

**Reply by Sophyia7** [reply]
![Screenshot 2022-07-17 192555](https://user-images.githubusercontent.com/66218403/179422781-e69b4683-557f-46b0-99df-eda92d62124f.png)
This is how it is on my root directory..how do I resolve the SSL cert issue? @fooforge

**Reply by fooforge** [reply]
@Sophyia7 No problem. To learn how to connect with Python, check out our [connection-examples repository](https://github.com/planetscale/connection-examples/tree/main/python). We have examples for Django, the [mysql-connector-python](https://dev.mysql.com/doc/connector-python/en) and [MySQLdb](https://mysqlclient.readthedocs.io/user_guide.html) adapters.

If you're running into any error messages that you cannot make sense of and fix on your own, drop them in here and we will take a look.

**Reply by Sophyia7** [reply]
I did use mysqlclient tho.

**Comment by jeremyikwuje** [comment]
I was able to successfully connect to the database by adding a path to the .env file:

```
HOST=myhost 
DATABASE=php_example
USERNAME=myusername
PASSWORD=pscale_password
MYSQL_ATTR_SSL_CA=cacert.pem
```

If this is not working, you are sure your database, host, username, and password are correct; then try changing the path base to the SSL. For instance, I Initially used /cacert.pem but it didn't work. Then I switched to the above.

PS: This is for a PHP application.

**Reply by Sophyia7** [reply]
So should I leave the cacert.pem where it is on the snapshot I shared above? Or should I open the file and copy the code into my .env file? 

If you don't mind, maybe we can hop on a call and can explain it better to me. Thanks in advance :)

**Reply by jeremyikwuje** [reply]
On the .env file, link to the base directory where the cacert.pem file is.

Fine. Send a link. Google Meet preferably.

**Reply by Sophyia7** [reply]
https://meet.google.com/hid-bdds-bro

**Reply by Sophyia7** [reply]
Let me know when you join. Thanks.

**Reply by jeremyikwuje** [reply]
I had to attend to something. Let me in.

**Reply by Sophyia7** [reply]
The sample stuff wasn't working for me so I want to figure an error before I try it again.

**Reply by jeremyikwuje** [reply]
Alright

**Reply by Credicxotech** [reply]
Hey @ijsucceed , just stuck into a problem ,  I want to connect a python file to one my pscale db, but it is continously throwing error- "MySQLdb.NotSupportedError: MySQL client library does not support ssl_mode specification", though I have downloaded the file from https://curl.se/docs/caextract.html, but still now working. Any idea what could have gone wrong ?

**Reply by Sophyia7** [reply]
Did you download the Cacert.pem file? If yes, add the path to the file in your root directory and add the path on your .env like the image below

![image](https://user-images.githubusercontent.com/66218403/183645248-f11a3bf7-a477-41c3-9676-e04957c6a948.png)

**Comment by Credicxotech** [comment]
I am using this code and the error is with library , new in this business so please tell me how would I know that I have made successful connection? It does not give me any error when I remove ssl_mode parameter from the fucntion.

![Screenshot 2022-08-09 180645](https://user-images.githubusercontent.com/93118326/183648626-d06ad489-f47e-4577-a2fb-8d0d4d328821.png)

**Reply by Sophyia7** [reply]
![image](https://user-images.githubusercontent.com/66218403/183652397-eabd9697-f6ae-4347-85a6-cfc33815edc2.png)

This is how I did mine.

**Reply by Credicxotech** [reply]
Ok , thanks, will do this
